server/controllers/nlp.js

```diff
@@ -47,12 +208,76 @@ exports.handleInput = function (req, res) {
 
         let dataset = req.body.dataset;
 
+        let lowerCaseInput = req.body.input.toLowerCase();
+
+        var tokenizer = new natural.WordTokenizer();
+        let tokenizedInput = tokenizer.tokenize(lowerCaseInput);
+
+
+        let initState = {
+            tokens: tokenizedInput,
+            layer: 0,
+            currentToken: 0,
+            callback: state => {
+                let bestMatch = {
+                    function: 'plotPieChartOfColumn',
+                    parameters: []
+                };
+                var numberMatches;
+                var numberColumns;
+                var columnsArray;
+                var bestMatched = {
+                    numberMatches: 0,
+                    function: "",
+                    functionParameter: []
+                };
+                _.forEach(commands, function (command) {
+                    numberMatches = 0;
+                    numberColumns = 0;
+                    columnsArray = [];
+                    _.forEach(state.tokens, function (token) {
+                        if (token.type === "Column") {
```

Comment:
could we use a static section where we define such strings only once and use them as reference from then on?
```
let static = {
   column: "Column",
   ...
}

...

if(token.type === static.column) {
...
```

Reply:
good idea, thanks!